AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -47,18 +52,24 @@ actor EndpointClient: EndpointClientBehaviour {
          pinpointClient: PinpointClientProtocol,
          archiver: AmplifyArchiverBehaviour = AmplifyArchiver(),
          currentDevice: Device = DeviceProvider.current,
-         userDefaults: UserDefaultsBehaviour = UserDefaults.standard) {
+         userDefaults: UserDefaultsBehaviour = UserDefaults.standard,
+         keychain: KeychainStoreBehavior = KeychainStore()
```

Comment:
Don't forget to pass the right instance when this client is created in [`PinpointContext`](https://github.com/aws-amplify/amplify-ios/blob/dev-preview/AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/PinpointContext.swift#L149).

Also, you might wanna add the proper `service` key to the default value:
```
KeychainStore(service: PinpointContext.Constants.Keychain.service)
```

Reply:
Updated constructor and also specify the service.